src/main/java/pl/allegro/tech/search/elasticsearch/tools/reindex/process/QueryComponent.java

```diff
@@ -36,12 +43,28 @@
   }
 
   public SearchResponse prepareSearchScrollRequest() {
+    // find out how many indices and shards are affected by this query to not get huge result sets when there are very many indices affected by the name, e.g. when wildcards are used
+    // otherwise we regularly run into OOMs when a query goes against a large number of indices
+    // I did not find a better way to find out the number of Shards then to get a list of indices and for each index query the replicas/shards via the settings
+    GetSettingsResponse getSettingsResponse = client.admin().indices().getSettings(new GetSettingsRequest().indices(dataPointer.getIndexName())).actionGet();
```

Comment:
Could you extract shardSize calculating into an other component and write a test for this method?